### PR TITLE
Query args parsing support

### PR DESCRIPTION
**Pull request by alfredodeza**

### CODE REVIEW

File: shaman/tests/test_util.py [hunk 1]
```diff
@@ -105,3 +106,54 @@ def test_pick_last_used_node(self, session, monkeypatch):
...
+class TestParseDistroRelease(object):
+
+    @py.test.mark.parametrize('version,codename', [('14.04', 'trusty'), ('16.04', 'xenial'), ('16.10', 'yakkety')])
+    def test_ubuntu_version_gets_both_parsed(self, version, codename):
+        c, v = util.parse_distro_release(version)
+        assert v == version
+        assert c == codename
+
+    @py.test.mark.parametrize('version,codename', [('14.04', 'trusty'), ('16.04', 'xenial'), ('16.10', 'yakkety')])
+    def test_ubuntu_codename_gets_both_parsed(self, version, codename):
```
Comment (andrewschoen) [inline shaman/tests/test_util.py]: How is this test any different than ``test_ubuntu_version_gets_both_parsed``?

File: shaman/util.py [hunk 2]
```diff
@@ -55,3 +55,54 @@ def is_node_healthy(node):
...
     models.commit()
     return True
+
+
+def parse_distro_release(identifier):
+    """
+    Back and forth translation for a release identifier, falling back to
+    ``None`` when an identifier has no existing mapping.
+
+    For CentOS, an identifier of '7' will return: None, '7'.
```
Comment (andrewschoen) [inline shaman/util.py]: I think the codename for CentOS is ``core``. Should we return that even though every CentOS release has the same codename?

File: shaman/util.py [hunk 3]
```diff
@@ -55,3 +55,54 @@ def is_node_healthy(node):
...
+
+    For an identifier of 'xenial', will return: 'xenial', '16.04'
+
+    returns: 2 item tuple (codename, version)
+    """
+    version_map = {
```
Comment (andrewschoen) [inline shaman/util.py]: This is obviously incomplete. Should we include debian, centos and rhel in this as well?

File: shaman/tests/test_util.py (hunk 1 above)
Comment (alfredodeza) [inline shaman/tests/test_util.py]: the first one passes the version, this one passes the codename. The results are the same

File: shaman/util.py (hunk 3 above)
Comment (alfredodeza) [inline shaman/util.py]: I can add Debian, but I am hesitant to add anything else (rpm-based distros mainly)

File: shaman/tests/test_util.py [hunk 4]
```diff
@@ -105,3 +106,68 @@ def test_pick_last_used_node(self, session, monkeypatch):
...
+        c, v = util.parse_distro_release(codename)
+        assert v == version
+        assert c == codename
+
+    def test_centos_gets_version_parsed(self):
+        codename, version = util.parse_distro_release('7')
+        assert version == '7'
+        assert codename is None
```
Comment (andrewschoen) [inline shaman/tests/test_util.py]: This test is now failing because wheezy and centos7 both have the version ``7``. :(

**Pull request merged**